src/editor/VideoViewer.js

```diff
@@ -0,0 +1,320 @@
+        Strings             = require("strings"),
+        Filer               = require("filesystem/impls/filer/BracketsFiler"),
+        Path                = Filer.Path,
+        StringUtils         = require("utils/StringUtils"),
+        FileSystem          = require("filesystem/FileSystem"),
+        UrlCache            = require("filesystem/impls/filer/UrlCache"),
+        FileUtils           = require("file/FileUtils"),
+        Content             = require("filesystem/impls/filer/lib/content"),
+        _                   = require("thirdparty/lodash"),
+        Mustache            = require("thirdparty/mustache/mustache"),
+        StartupState        = require("bramble/StartupState");
+
+    var _viewers = {};
+
+    // Get a Blob URL out of the cache
+    function _getVideoUrl(file) {
+        return UrlCache.getUrl(file.fullPath);
+    }
+
+    // Get a URL out of the cache that you can use in the project HTML
+    function _getLocalVideoUrl(file) {
+        var fullUrl = UrlCache.getFilename(UrlCache.getUrl(file.fullPath));
+        var root = StartupState.project("root");
+        return fullUrl.replace(root,"").replace("/","");
+    }
+
+    /**
+     * Check if it's a video file TODO: refactor
+     */
+    function isVideo(fullPath) {
+        return Content.isVideo(Path.extname(fullPath));
+    }
+
+    /**
+     * VideoView objects are constructed when a video is opened
+     * @see {@link Pane} for more information about where VideoViews are rendered
+     *
+     * @constructor
+     * @param {!File} file - The video file object to render
+     * @param {!jQuery} container - The container to render the video view in
+     */
+    function VideoView(file, $container) {
+        var that = this;
+
+        this.file = file;
+        this.container = $container;
+
+        // Set defaults, and keep track of which video element attributes are enabled.
+        this.videoTagSettings = {
+            controls: true,
+            autoplay: false,
+            loop: false,
+            muted: false
+        };
+
+        // Gets the video type for sample markup
+        this.videoType = Content.mimeFromExt(Path.extname(this.file.fullPath));
+
+        this.$el = $(Mustache.render(VideoViewTemplate, {
+            videoUrl: _getVideoUrl(file),
+            videoType: this.videoType,
+            Strings: Strings
+        }));
+
+        $container.append(this.$el);
+
+        this._naturalWidth = 0;
+        this._naturalHeight = 0;
+
+        this.relPath = ProjectManager.makeProjectRelativeIfPossible(this.file.fullPath);
+
+        this.$videoEl = this.$el.find("video");
+        this.$videoWrapperEl = this.$el.find(".video-wrapper");
+        this.$videoMarkupEl = this.$el.find("pre");
+        this.$videoData = this.$el.find(".video-data");
+
+        this.$videoEl.one("canplay", _.bind(this._onVideoLoaded, this));
+        this.$videoEl.on("error", _.bind(console.error, console));
```

Comment:
I think we can upgrade this view later to also remember which options you selected for the sample markup, but for now I think it's okay if it resets every time.